docs/manual/hacking.rst

```diff
@@ -153,3 +162,89 @@ Here are a number of resources that may come in handy:
 * `Inter-Client Conventions Manual <http://tronche.com/gui/x/icccm/>`_
 * `Extended Window Manager Hints <http://standards.freedesktop.org/wm-spec/wm-spec-latest.html>`_
 * `A reasonable basic Xlib Manual <http://tronche.com/gui/x/xlib/>`_
+
+
+Troubleshoot
+============
+
+Cairo errors
+------------
+
+When running the Xephyr script (``./scripts/xephyr``), you might see tracebacks
+with attribute error like the following or similar::
+
+    AttributeError: cffi library 'libcairo.so.2' has no function, constant or global variable named 'cairo_xcb_surface_create'
+
+If it happens, it might be because the `cairocffi` and `xcffib` dependencies
+were installed in the wrong order.
+
+To fix this:
+
+1. uninstall them from your environment: with ``pip uninstall cairocffi xcffib``
+   if using a virtualenv, or with your system package-manager if you installed
+   the development version of Qtile system-wide.
+#. re-install them sequentially (again, with pip or with your package-manager)::
```

Comment:
If it is a package manager, it may be up to fixing the packaging of cairocffi in the relevant distro.

Reply:
I added this as a note just below.